### PR TITLE
Feature/setup

### PR DESCRIPTION
**Pull request by 0sugo**
THe following were the project requirements:

- Splash screen

    - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
    Sign up and log in pages

- The user should be able to register in the app with full name, email and password (all mandatory).

    - The user can log into the app using email and password.
    - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)

    - When the user logs in, they are presented with the categories page.
    - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
    - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
    - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page

    - For a given category, the list of transactions is presented, ordered by the most recent.
    - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page

    - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
    - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

    - The user fills out a form to create a new transaction with:
        - name (mandatory)
        - amount (mandatory)
        - categories (mandatory at least one)
    - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
    - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- Testing requirements
    - Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: app/models/financial_group.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class FinancialGroup < ApplicationRecord
+  belongs_to :user
+  has_many :categorisations, dependent: :destroy
+  has_many :financial_entities, through: :categorisations
+  has_one_attached :icon
+  validates :name, presence: true
```
Comment (danmainah) [inline app/models/financial_group.rb]: - [ ] @0sugo Great work adding validation for the name kindly add the same for the icon, its part of te project requirement
image

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  # get 'home', to: 'home#index', as: 'home'
+  devise_for :users
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+
+  # Defines the root path route ("/")
+  root "home#index"
+
+  resources :financial_groups do
+  resources :financial_entities
+  end
```
Comment (danmainah) [inline config/routes.rb]: - [ ] @0sugo Great work on creating your routes, however, you have some routes that are unnecessary, like for updating and deleting categories and entities yet they are not being used.

File: spec/models/financial_group_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      financial_group = FinancialGroup.new(icon: file)
+      expect(financial_group).to_not be_valid
+      expect(financial_group.errors[:name]).to include("can't be blank")
+    end
+
+    # it 'is invalid without an icon' do
+    #   financial_group = FinancialGroup.new(name: 'Sample Group')
+    #   expect(financial_group).to be_valid
+    #   expect(financial_group.errors[:icon]).to include("can't be blank")
+    # end
```
Comment (danmainah) [inline spec/models/financial_group_spec.rb]: - [ ] @0sugo It's not good practice to have commented code in your codebase, kindly remove this test or ensure its test is passing.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @0sugo , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added the presentation showcasing the project's features.
✔️  Ensured the pages follow the design given.
✔️  Ensured all projects required functionalities are working.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Whoistolu — APPROVED** [review]
### Hi @0sugo 

Great job implementing the recommendations by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**